### ISSUE TITLE
XMLRPC does not seem to work at all

### ISSUE THREAD

**Issue by mpedrummer** [jakoch/PHPTracRPC#15]
Trying to troubleshoot some problems, I my URL from `login/jsonrpc` to `login/xmlrpc`, mainly grasping at straws.  Unfortunately, `doRequest()` will `json_encode()` ALL requests, regardless of content type.

This results in an XML response of:

    <?xml version='1.0'?> 
    <methodResponse> 
      <fault> 
        <value>
          <struct> 
            <member> 
              <name>faultCode</name> 
              <value><int>-32700</int></value> 
            </member> 
            <member> 
              <name>faultString</name> 
              <value><string>not well-formed (invalid token): line 1, column 0</string></value> 
            </member> 
          </struct>
        </value> 
      </fault> 
    </methodResponse>

Which makes some sense, because it was passed a json-encoded string.

Compounding the problem, `$this->error` was not set.

I'm not sure how to fix this one.  From my perspective, I'd just drop xmlrpc support, and simplify the class by using jsonrpc only.  Since the class should be normalizing the input and output for Trac to be native PHP, I don't see significant value in supporting both, or at least not enough to justify the extra work required at this point to implement xmlrpc.

**Comment by jakoch** [comment]
I think, when the request is made with the correct content-type (xml) 
then it might work by using `xmlrpc_decode(trim($response));`
but yes, not implemented.

**Comment by jakoch** [comment]
Ok, i've implemented basic XMLRPC handling.
Tested with a single request to getApiVersion for now.

Todo:
- [ ] add multicall support and store the answers with ids
- [ ] cleanup parseResult() function
   - maybe add two new classes to handle the responses properly: `JsonResponse()` `XmlResponse()`

**Comment by mpedrummer** [comment]
This looks like a good start!  I think your suggestion to refactor into Response classes is wise - perhaps doing the same for Request classes, too.  It'd make the execution path easier to understand, too - fewer conditional paths to mentally parse.

Yay tests!

**Comment by jakoch** [comment]
> perhaps doing the same for Request classes

I agree. We could introduce `JsonRequest` and `XmlRequest` with `CurlRequest` as base class.
